crates/nu-path/Cargo.toml

```diff
@@ -16,5 +16,8 @@ dirs-next = "2.0"
 [target.'cfg(windows)'.dependencies]
 omnipath = "0.1"
 
-[target.'cfg(all(unix, not(target_os = "macos")))'.dependencies]
+[target.'cfg(all(unix, not(target_os = "macos"), not(target_os = "android")))'.dependencies]
 pwd = "1.3"
+
+[dev-dependencies]
+whoami = "1.4.1"
```

Comment:
Well the problem is the function needs a user name as input. I could just give it a dummy value, but then it never actually hits the code path that gets the home dir from the system, and essentially always uses the fallback, which I think kind of defeats the point of the test. (It's just a coincidence that the correct value that we get from the OS is the same as the answer when it falls back.)

On another note, while I was making this change, I found myself wondering if this "fallback" behavior really makes sense to do at all. I'm not sure it makes much sense to give back a non-existent home directory, rather than just being fallible. But even if this were agreed, that would involve a potentially much bigger change.

Reply:
Ah, I forgot you need the username. You can use `dirs_next::home_dir()` instead of `whoami` and treat the last path element as a username. That is already used in this crate.

Yeah, we could look into the fallback mechanism, but I'd leave that out of this PR.

Reply:
> Ah, I forgot you need the username. You can use `dirs_next::home_dir()` instead of `whoami` and treat the last path element as a username. That is already used in this crate.

But that is not correct. The home directory is not guaranteed to be `/home/<username>`, that is just the default if you do not specify the home dir when you make the user. It's quite common e.g. to make dedicated users for running a service as non-root with a home in `/var/lib/something`. If this crate is assuming the last component is always the user name, then it would be quite interesting to see what happens when someone tried to run nushell as such a user.

The only reliable way to get the home directory is to ask the host operating system. We could reach into `nix` or `winapi` to do this, but that would require more conditional code, and at that point, we're just reimplementing `whois`.

Actually the more I think about it, the less I feel a unit test is appropriate for this. Assuming the "correct" home dir is `FALLBACK/username` will fail when anyone runs the tests on a system who has set a nonstandard home directory.

Reply:
OK, TIL. So maybe we can just test that the fallback points at the right directory.

Reply:
Yeah, we can at least test the fallback behavior until either that fallback mechanism is changed or we have some mechanism for making test environments.

Reply:
Ok this is done